CHANGELOG.md

```diff
@@ -1,5 +1,63 @@
 # go-ipfs changelog
 
+### 0.4.2 - 2016-05-17
+
+This is a patch release which fixes perfomance and networking bugs in go-libp2p,
+You should see improvements in CPU and RAM usage, as well as speed of object lookups.
+There are also a few other nice improvements.
+
+* Notable Fixes
+  * Set a deadline for dialing attempts. This prevents a node from accumulating
+    failed connections. (@whyrusleeping)
+  * Avoid unneccessary string/byte conversions in go-multihash. (@whyrusleeping)
+  * Fix a deadlock around the yamux stream muxer. (@whyrusleeping)
```

Comment:
The entries here often don't match one-to-one with a PR -- but I agree that for the interesting changes it'd be pretty useful.

Next time? :) Or I can give it another go tomorrow.